### ISSUE TITLE
GetFrameTimings not works in c#?

### ISSUE THREAD

**Issue by Zaetc** [ValveSoftware/openvr#1001]
GetFrameTiming works fine. (it ok, but it is not optimal way if you need get many Compositor_FrameTiming)
But GetFrameTiming**s** brings to crashes (System.AccessViolationException).
May be I'm doing something wrong?
Here is code:
```
 private Compositor_FrameTiming[] GetFrames2()
        {
            Compositor_FrameTiming[] framestimesBuffer = new Compositor_FrameTiming[40];
            framestimesBuffer[0].m_nSize = Convert.ToUInt32(System.Runtime.InteropServices.Marshal.SizeOf(typeof(Compositor_FrameTiming)));
            uint ft = OpenVR.Compositor.GetFrameTimings(ref framestimesBuffer[0], 40);
            if (ft != 40)
            {
                /// do sometning
            }
            return framestimesBuffer;
        }
```

**Comment by BOLL7708** [comment]
I noticed this as well, so I just went with grabbing the frame timing very often instead, as I'm doing live frame timing analysis. It's a while since I looked at this, I think loading timings worked if I only asked for 1-2 frames, which made it kind of pointless, if I actually remember it correctly.

**Comment by aleiby** [comment]
That function is missing the proper mark up to have it export properly in the C# bindings.  Here's what you can do to change this locally until we get a new version of the SDK updated with the fix.

```
[UnmanagedFunctionPointer(CallingConvention.StdCall)]
internal delegate uint _GetFrameTimings([In, Out] Compositor_FrameTiming[] pTiming, uint nFrames);
[MarshalAs(UnmanagedType.FunctionPtr)]
internal _GetFrameTimings GetFrameTimings;

public uint GetFrameTimings(Compositor_FrameTiming [] pTiming)
{
	uint result = FnTable.GetFrameTimings(pTiming,(uint) pTiming.Length);
	return result;
}

```
Then you just call like:
```
var timing = new Compositor_FrameTiming[8];
timing[0].m_nSize = (uint)System.Runtime.InteropServices.Marshal.SizeOf(typeof(Compositor_FrameTiming));
vr.compositor.GetFrameTimings(timing);

```

**Comment by aleiby** [comment]
Should be fixed in the next SDK update.  Thanks!

**Comment by BOLL7708** [comment]
Thanks for replying here @aleiby, this got me to post about [something else](https://github.com/ValveSoftware/openvr/issues/1016) in the C# header I've been thinking about for a long time now 😅 I have a few other things regarding the header as well that I'll try to post eventually, need to do some more testing first though.

**Comment by BOLL7708** [comment]
@aleiby I finally got to updating my implementation, I'm providing an array `Compositor_FrameTiming[]` and the result count I get back matches the array size so it would appear to be working, but then all objects in the array are still empty (just instantiated with default primitives). Am I missing something here? 🤔 Meanwhile I can successfully get data from `GetFrameTiming()`.

Could it be that the array reference is replaced while not passed as reference? I tried adding the key word but no difference on my end, so not sure. Did make a [fiddle](https://dotnetfiddle.net/hioXb7) to check my sanity.
